src/util/strencodings.h

```diff
@@ -71,9 +71,20 @@ std::vector<Byte> ParseHex(std::string_view hex_str)
  * number of hex digits.*/
 bool IsHex(std::string_view str);
 /**
-* Return true if the string is a hex number, optionally prefixed with "0x"
-*/
-bool IsHexNumber(std::string_view str);
+ * @brief Return a sanitized hex string if input is a valid hex number.
+ *        "0x" prefix is removed, and if result_size is specified, the
+ *        result is padded with leading zeroes until result_size is
+ *        reached, or std::nullopt returned if input is longer than
+ *        result_size.
+ *
+ * @param input Hex encoding of a number, optionally prefixed with 0x.
+ * @param result_size Size of the resulting string, causing zero-padding
```

Comment:
I think we could document how to disable this param - currently with 0, I recommend considering negative numbers